### DISCUSSION TITLE
WtLocalExecuter和WtDistExecuter这两个执行器的功能职责是什么？

### DISCUSSION THREAD

**Discussion by Praying** [wondertrader/wondertrader#51] [Q&A] [answered]
在WtCtaEngine和WtSelEngine中分别有一个名为_exec_mgr的成员，其类型为WtExecuterMgr，在该类中维护一个IExecCommand的vector，其中WtLocalExecuter和WtDistExecuter分别是IExecCommand的具体实现，所以问题是，WtLocalExecuter和WtDistExecuter这两个执行器的功能职责是什么？

**Comment by wondertrader** [comment, marked as answer]
WonderTrader的CTAEngine和SELEngine采用的是信号和执行分离的架构。在同一个进程中，可以配置多个执行器，同样WonderTrader也支持分布式执行。这里的WtDistExecuter，完整名称为WtDistributedExecuter，是分布式执行器的意思，但是目前只是留了一个口子，以后会采用sharememory的方式来实现，这样就能做到信号和执行在不同进程中处理的目的了。
